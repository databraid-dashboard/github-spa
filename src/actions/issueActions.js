// import { issuesMock } from '../SampleJSONResponses/IssuesJSON';

export const GET_ISSUES = 'GET_ISSUES';
export const LOADING_ISSUES = 'LOADING_ISSUES';

export function retrieveIssues(userName, orgName, repoName) {
  const queryString = `{repos (userName:"${userName}", orgName: "${orgName}", repoName:"${repoName}") {repos {issues {id title number state labels assignee assigneeAvatar } } } }`;

  const request = { query: queryString };

  return (dispatch, getState, { Api }) => Api.fetchData(request)
    .then(response => response.data.repos.repos[0].issues)
    .then((issues) => {
      dispatch({
        type: GET_ISSUES,
        responseObj: issues,
        repoName,
      });
    });
}
