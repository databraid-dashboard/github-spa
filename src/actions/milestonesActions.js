export const GET_MILESTONES = 'GET_MILESTONES';
export const LOADING_MILESTONES = 'LOADING_MILESTONES';

export function retrieveMilestones(userName, orgName, repoName) {
  const queryString = `
    {
      repos (
        userName: "${userName}",
        orgName:"${orgName}",
        repoName:"${repoName}"){
          repos {
            milestones {
              id
              title
              openIssues
              closedIssues
              dueOn
            }
          }
        }
      }
      `;

  const request = { query: queryString };

  return (dispatch, getState, { GITHUB_API }) =>
    GITHUB_API.fetchData(request)
      .then(response => response.data.repos.repos[0].milestones)
      .then((milestones) => {
        dispatch({
          type: GET_MILESTONES,
          responseObj: milestones,
          repoName,
        });
      });
}
