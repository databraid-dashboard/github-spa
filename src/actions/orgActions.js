import authorizedUserByOrgResponse from '../SampleJSONResponses/authORGsJSON';

export const GET_ORGS = 'GET_ORGS';

export const retrieveOrgs = () => (dispatch) => {
  dispatch({
    type: GET_ORGS,
    responseObj: authorizedUserByOrgResponse,
  });
};
