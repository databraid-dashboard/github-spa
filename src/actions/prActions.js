// import prMock from '../SampleJSONResponses/PRmock';

export const GET_PRS = 'GET_PRS';
export const LOADING_PRS = 'LOADING_PRS';

export function retrievePrs(userName, orgName, repoName) {
  const queryString = `{repos (userName: "${userName}", orgName:"${orgName}", repoName:"${repoName}") {repos {pullRequests {id title submittedBy createdAt mergeable } } } }`;

  const request = { query: queryString };

  return (dispatch, getState, { Api }) => Api.fetchData(request)
    .then(response => response.data.repos.repos[0].pullRequests)
    .then((pullRequests) => {
      dispatch({
        type: GET_PRS,
        responseObj: pullRequests,
        repoName,
      });
    });
}
