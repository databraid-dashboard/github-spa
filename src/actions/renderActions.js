
export const DISPLAY_ORGS = 'DISPLAY_ORGS';
export const DISPLAY_LOGIN = 'DISPLAY_LOGIN';
export const DISPLAY_REPOS = 'DISPLAY_REPOS';
export const LOGOUT = 'LOGOUT';
export const DISPLAY_DASHBOARD = 'DISPLAY_DASHBOARD';

export const displayOrgs = userName => (dispatch) => {
  dispatch({
    type: DISPLAY_ORGS,
    userName,
  });
};

export const displayRepos = selectedOrgName => (dispatch) => {
  dispatch({
    type: DISPLAY_REPOS,
    selectedOrgName,
  });
};

export const displayDashboard = repoName => (dispatch) => {
  dispatch({
    type: DISPLAY_DASHBOARD,
    repoName,
  });
};
export const logout = () => ({
  type: LOGOUT,
});
