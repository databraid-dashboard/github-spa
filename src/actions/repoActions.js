import reposByOrg from '../SampleJSONResponses/reposByORGmock';

export const GET_REPOS = 'GET_REPOS';

export const retrieveRepos = () => (dispatch) => {
  dispatch({
    type: GET_REPOS,
    responseObj: reposByOrg,
  });
};
