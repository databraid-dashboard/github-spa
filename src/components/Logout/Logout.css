.butterButton {
  height:75px!important;
  width:300px!important;
}

.island {
  margin-top:20%!important;
  height: 105px!important;
  width:330px!important;
}
