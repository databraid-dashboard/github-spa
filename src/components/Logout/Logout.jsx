import React from 'react';
import { Grid, Card, Button } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import cookie from 'react-cookies';
import PropTypes from 'prop-types';
import './Logout.css';
import { logout } from '../../actions/renderActions';
/* eslint-disable no-shadow */
export const triggerLogout = (logout) => {
  cookie.remove('userName');
  cookie.remove('githubAccessToken');
  logout();
};

export const Logout = ({ logout }) => (
  /* eslint-enable */
  <Grid verticalAlign={'top'} >
    <Grid.Row >
      <Grid.Column verticalAlign={'middle'}>
        <Card.Content>
          <Button
            as="a"
            onClick={() => { triggerLogout(logout); }}
            className="hand peace icon"
            size="medium"
            content="Logout"
            icon="hand peace"
          />
        </Card.Content>
      </Grid.Column>
    </Grid.Row>
  </Grid>
  );


Logout.propTypes = {
  /* eslint-disable react/no-unused-prop-types, react/require-default-props */
  logout: PropTypes.func.isRequired,
};
export const mapDispatchToProps = dispatch => bindActionCreators({
  logout,
}, dispatch);

export default connect(null, mapDispatchToProps)(Logout);
