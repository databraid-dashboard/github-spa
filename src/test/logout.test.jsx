import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { Logout, mapDispatchToProps } from '../components/Logout/Logout';
import { logout } from '../actions/renderActions';


const logoutComponent = shallow(
  <Logout
    logout={logout}
  />,
);


describe('Logout Component', () => {
  it('renders a button component as specified', () => {
    expect(toJson(logoutComponent)).toMatchSnapshot();
  });
  it('should have a length of 1', () => {
    expect(logoutComponent.toHave);
  });
  it('should contain an icon', () => {
    expect(logoutComponent.find('.icon').exists()).toBe(true);
  });
  it('should contain an icon, the icon should be a hand', () => {
    expect(logoutComponent.find('.hand').exists()).toBe(true);
  });
  it('maps logout dispatches to props', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('logout');
  });
});
