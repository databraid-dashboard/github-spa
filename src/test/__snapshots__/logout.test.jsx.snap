// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Component renders a button component as specified 1`] = `
<Grid
  verticalAlign="top"
>
  <GridRow>
    <GridColumn
      verticalAlign="middle"
    >
      <CardContent>
        <Button
          as="a"
          className="hand peace icon"
          content="Logout"
          href="undefined"
          icon="hand peace"
          size="medium"
        />
      </CardContent>
    </GridColumn>
  </GridRow>
</Grid>
`;
